file-events/src/file-events/cpp/linux_fsnotifier.cpp

```diff
@@ -119,9 +105,7 @@ void Server::processQueues(int timeout) {
     }
 
     if (IS_SET(fds[0].revents, POLLIN)) {
-        shutdownEvent.consume();
-        // Ignore counter, we only care about the notification itself
-        shouldTerminate = true;
+        commandEvent.process();
```

Comment:
Isn't there a race condition here? `shutdownRunLoop` could run while this thread here is processing the command. That would mean that the command in `commandEvent` could be modified while this runLoop is processing the command in `commandEvent`. Or am I misunderstanding something here?

Can you add a description to the PR what kind of race condition we are trying to fix here?

Reply:
`FileWatcher` is supposed to be accessed from a single thread, it is marked `@NotThreadSafe` because of this reason. This means we don't need to prepare for multiple commands to be executed at the same time. Is there any additional description you'd like to add?